src/components/shared/mixins/urlParams.js

```diff
@@ -0,0 +1,25 @@
+export default {
+    methods: {
+        urlParamGen(paramObject, prefix = '', delimeter = '&') {
+            const str = []
+            for (const key in paramObject) {
+                switch (typeof paramObject[key]) {
+                case 'object':
+                    if (Array.isArray(paramObject[key])) {
+                        paramObject[key].forEach((param) => {
+                            str.push(`${encodeURIComponent(prefix.length ? `${prefix}[${key}][]` : `${key}[]`)}=${encodeURIComponent(`${param}`)}`)
+                        })
+                    } else {
+                        str.push(this.urlParamGen(paramObject[key], key))
```

Comment:
I think this needs to be prefix aware, else this will only work for root objects.

Off the top of my head, I think something like this should work.
```js
str.push(this.urlParamGen(paramObject[key], prefix.length ? `${prefix}[${key}]` : `${key}`))
```

Could we add some test coverage for this too please?

Reply:
Thanks, i have update the method with the missing logic i also noticed it didnt handle objects as array values so also added that in (thanks to the tests :D )

Reply:
As in an array of objects? As I don't think this does handle that properly.
`query[deepObject][evenDeeper][assets][channels][id]=1` is not correct.

Shouldn't it be something like...`query[deepObject][evenDeeper][assets][channels][0][id]=1`

Reply:
yeah an array of objects, ah gotcha i have added in the index. Cheers

Reply:
Cool, how are you generating the expected results for the tests?

Reply:
I give a dud at first in the expected result and take the received, and manually check it decoded to ensure it looks correct before setting it as the expected output.

Reply:
Ok, that all looks good now, but just for our piece of mind and to remove the human error in the manual method above, can we check the output against something like the jQuery param method?

```js
import { param } from 'jquery'
...
expect(vm.urlParamGen(exportPaths[path].params)).toBe(param(exportPaths[path].params))
```

If we are constrained by time, I am happy for this to be a tech debt task and I can publish a prerelease to use while we wait?

Reply:
Thats a good idea and not to big of a change so i have just updated the tests. Thanks